sycl/test/usm/allocator_vector.cpp

```diff
@@ -30,35 +30,106 @@ int main() {
   auto dev = q.get_device();
   auto ctxt = q.get_context();
 
-  if (!dev.get_info<info::device::usm_host_allocations>()) 
+  if (!(dev.get_info<info::device::usm_host_allocations>() &&
+        dev.get_info<info::device::usm_device_allocations>() &&
+        dev.get_info<info::device::usm_shared_allocations>()))
     return 0;
 
-  usm_allocator<int, usm::alloc::host> alloc(ctxt, dev);
+  {
+    usm_allocator<int, usm::alloc::host> alloc(ctxt, dev);
 
-  std::vector<int, decltype(alloc)> vec(alloc);
-  vec.resize(N);
+    std::vector<int, decltype(alloc)> vec(alloc);
+    vec.resize(N);
 
-  for (int i = 0; i < N; i++) {
-    vec[i] = i;
+    for (int i = 0; i < N; i++) {
+      vec[i] = i;
+    }
+
+    int *res = &vec[0];
+    int *vals = &vec[0];
+
+    auto e1 = q.submit([=](handler &h) {
+      h.single_task<class foo>([=]() {
+        for (int i = 1; i < N; i++) {
+          res[0] += vals[i];
+        }
+      });
+    });
+
+    e1.wait();
+
+    int answer = (N * (N - 1)) / 2;
+
+    if (vec[0] != answer)
+      return -1;
   }
 
-  int *res = &vec[0];
-  int *vals = &vec[0];
+  {
+    usm_allocator<int, usm::alloc::shared> alloc(ctxt, dev);
 
-  auto e1 = q.submit([=](handler &cgh) {
-    cgh.single_task<class foo>([=]() {
-      for (int i = 1; i < N; i++) {
-        res[0] += vals[i];
-      }
+    std::vector<int, decltype(alloc)> vec(alloc);
+    vec.resize(N);
+
+    for (int i = 0; i < N; i++) {
+      vec[i] = i;
+    }
+
+    int *res = &vec[0];
+    int *vals = &vec[0];
+
+    auto e1 = q.submit([=](handler &h) {
+      h.single_task<class bar>([=]() {
+        for (int i = 1; i < N; i++) {
+          res[0] += vals[i];
+        }
+      });
     });
-  });
 
-  e1.wait();
+    e1.wait();
 
-  int answer = (N * (N - 1)) / 2;
+    int answer = (N * (N - 1)) / 2;
+
+    if (vec[0] != answer)
+      return -1;
+  }
 
-  if (vec[0] != answer)
-    return -1;
+  {
+    usm_allocator<int, usm::alloc::device> alloc(ctxt, dev);
+
+    std::vector<int, decltype(alloc)> vec(alloc);
+    vec.resize(N);
+
+    int *res = &vec[0];
+    int *vals = &vec[0];
+
+    auto e0 = q.submit([=](handler &h) {
+      h.single_task<class baz_init>([=]() {
```

Comment:
Could be `parallel_for` here, to exercise more code paths?

Reply:
Other tests handle that.